make/go/scripts/checknolintlint.bash

```diff
@@ -0,0 +1,58 @@
+#!/usr/bin/env bash
+
+# Managed by makego. DO NOT EDIT.
+
+## checknolintlint exits with an exit code of 0 if nolintlint is enabled and configured properly.
+## It exits with an exit code of 2 if nolintlint is not enabled in .golangci.yml.
+## It exits with an exit code of 1 if nolintlint is not configured according to standards.
+
+set -euo pipefail
+
+# Check if nonolint linter is enabled in config
+NOLINTLINT_ENABLED=0
+if yq '.linters.enable // [] | any_c(. == "nolintlint")' .golangci.yml > /dev/null 2>&1; then
+    # Enabled individually
+    NOLINTLINT_ENABLED=1
+elif yq --exit-status '.linters.enable-all' .golangci.yml > /dev/null 2>&1; then
+    # Enabled with enable-all
+    NOLINTLINT_ENABLED=1
+fi
+if [ "${NOLINTLINT_ENABLED}" -eq 1 ]; then
+    # Ensure it isn't disabled individually
+    if yq --exit-status '.linters.disable // [] | any_c(. == "nolintlint")' .golangci.yml > /dev/null 2>&1; then
+        NOLINTLINT_ENABLED=0
+        echo "disabled individually" >&2
+    fi
+fi
+if [ "${NOLINTLINT_ENABLED}" -eq 0 ]; then
+    echo "not enabled" >&2
+    exit 2
+fi
+
+# Check if nolintlint is configured according to standards.
+#
+#   linters-settings:
+#     nolintlint:
+#       allow-unused: false
+#       allow-no-explanation: []
+#       require-explanation: true
+#       require-specific: true
```

Comment:
@bufdev - Confirmed with @rodaine that these were the agreed upon standards for projects that wish to use `//nolint:` comments.